src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -4041,7 +4056,12 @@ void StorageReplicatedMergeTree::dropPartition(const ASTPtr &, const ASTPtr & pa
     }
 
     /// Cleaning possibly stored information about parts from /quorum/last_part node in ZooKeeper.
-    cleanLastPartNode(partition_id);
+    /// TODO(nv) how is this related to dropPart? Is it?
```

Comment:
i believe i still need to handle this, but not sure how yet

Reply:
Here I suspect we need to do the following:

In a while loop:
1. Check if part is last_part for partition, if it is not: stop
2. If it is then we need to update `quorum/last_part` to be set to previous part for which quorum was achieved.
  Per my understanding since the blocks ranges in quorums are monotonic increasing we can get away with just decrementing max block by one? Would this work? What if quorum was achieved for blocks 1_1, 2_2, 3_3 and then 2_2 is deleted and after that 3_3? We would need to set quorum to 1_1 in this case? How do we do it? Do we need to keep full history of quorums for that?

What other alternatives exist?

I was thinking that we could find a previously (sorted by block numbers) committed part, but there are no guarantees that current replica is up to date? sync the queue (using ReplicatedMergeTreeMergePredicate again) and find a fake part name from virtual_parts instead?

Reply:
Another problem:
Let's say current quorum is for maxBlock = 2.
We insert DROP_RANGE to the queue and rewind the quorum/last_part to maxBlock = 1. All good till here.
Now a concurrent insert updates maxBlock to = 3, if all replicas did not process MAX_RANGE yet then some data will reappear until DROP_RANGE is executed...

Reply:
I think we can just restrict to drop part from `quorum/last_part`.